portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,35 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    public ArrayList<String> comment_list = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    comment_list = new ArrayList<String>();
+
+    String json = convertToJsonUsingGson(comment_list);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public String convertToJsonUsingGson(ArrayList<String> m) {
+      Gson gson = new Gson();
+      String json = gson.toJson(m);
+      return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    comment_list.add(text);
+
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("New Comment: " + text);
+    for (int i = 1; i < comment_list.size(); i++) {
+        response.getWriter().println("Comment: " + comment_list.get(i));
```

Comment:
You might want to add a new line after each comment for better display. Same for "New Comment" section above.